src/blockcrawler/evm/rpc.py

```diff
@@ -195,6 +195,77 @@ async def get_block(self, block_num: HexInt, full_transactions: bool = False) ->
         )
         return block
 
+    async def get_block_by_timestamp(self, timestamp: int) -> int:
```

Comment:
On further review, return the actual block instead of the block number. Getting a block number from a block is trivial. Getting a block from a block number would require an unnecessary call now since the block is retrieved in this process.